python/cudf/cudf/core/column/timedelta.py

```diff
@@ -484,88 +484,57 @@ def components(self) -> dict[str, ColumnBase]:
         4     37     13       12       14           234             0            0
         """
 
+        @functools.cache
+        def get_unit_conversion(reso: str) -> np.timedelta64:
```

Comment:
It seems like we can reuse this function throughout this module.

Reply:
Good idea. Was able to move this to the top level of this file